### ISSUE TITLE
Assertion Failed: (ec == 0), function unlock

### ISSUE THREAD

**Issue by joshwatson** [Vector35/binaryninja-api#1260]
I don't remember what i was doing for this one, or if it's reproducible. Was on 1.1.1486-dev:
```
Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Application Specific Information:
Assertion failed: (ec == 0), function unlock, file /BuildRoot/Library/Caches/com.apple.xbs/Sources/libcxx/libcxx-400.9.4/src/mutex.cpp, line 48.

Thread 0:: Dispatch queue: com.apple.main-thread
0   org.qt-project.QtGui          	0x0000000103483eb1 0x1030cb000 + 3903153
1   org.qt-project.QtGui          	0x0000000103483923 0x1030cb000 + 3901731
2   org.qt-project.QtGui          	0x00000001031d04ba QTextEngine::shapeTextWithHarfbuzzNG(QScriptItem const&, unsigned short const*, int, QFontEngine*, QVector<unsigned int> const&, bool, bool) const + 1146
3   org.qt-project.QtGui          	0x00000001031cedda QTextEngine::shapeText(int) const + 2394
4   org.qt-project.QtGui          	0x00000001031e7482 QTextLine::layout_helper(int) + 1586
5   org.qt-project.QtGui          	0x0000000103340ab5 0x1030cb000 + 2579125
6   org.qt-project.QtGui          	0x0000000103343a58 0x1030cb000 + 2591320
7   org.qt-project.QtGui          	0x00000001031bbd34 QFontMetrics::boundingRect(QRect const&, int, QString const&, int, int*) const + 212
8   com.vector35.binaryninja      	0x00000001004abb88 0x100468000 + 277384
9   org.qt-project.QtWidgets      	0x0000000100e3389a 0x100b82000 + 2824346
10  org.qt-project.QtWidgets      	0x0000000100e36136 0x100b82000 + 2834742
11  org.qt-project.QtWidgets      	0x0000000100e35f22 0x100b82000 + 2834210
12  org.qt-project.QtWidgets      	0x0000000100e2f48a 0x100b82000 + 2806922
13  org.qt-project.QtWidgets      	0x0000000100e332fe QListView::doItemsLayout() + 206
14  org.qt-project.QtWidgets      	0x0000000100dfc54a QAbstractItemView::timerEvent(QTimerEvent*) + 490
15  org.qt-project.QtCore         	0x000000010382c576 QObject::event(QEvent*) + 102
16  org.qt-project.QtWidgets      	0x0000000100bceaba QWidget::event(QEvent*) + 4746
17  org.qt-project.QtWidgets      	0x0000000100c74edd QFrame::event(QEvent*) + 45
18  org.qt-project.QtWidgets      	0x0000000100df7ca2 QAbstractItemView::event(QEvent*) + 546
19  org.qt-project.QtWidgets      	0x0000000100b91f7d QApplicationPrivate::notify_helper(QObject*, QEvent*) + 269
20  org.qt-project.QtWidgets      	0x0000000100b93382 QApplication::notify(QObject*, QEvent*) + 594
21  org.qt-project.QtCore         	0x0000000103802fb4 QCoreApplication::notifyInternal2(QObject*, QEvent*) + 212
22  org.qt-project.QtCore         	0x000000010385c2ef QTimerInfoList::activateTimers() + 991
23  libqcocoa.dylib               	0x0000000105824332 0x1057f8000 + 181042
24  com.apple.CoreFoundation      	0x00007fff32a93405 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
25  com.apple.CoreFoundation      	0x00007fff32a933ab __CFRunLoopDoSource0 + 108
26  com.apple.CoreFoundation      	0x00007fff32a76e51 __CFRunLoopDoSources0 + 195
27  com.apple.CoreFoundation      	0x00007fff32a763fa __CFRunLoopRun + 1219
28  com.apple.CoreFoundation      	0x00007fff32a75ce4 CFRunLoopRunSpecific + 463
29  com.apple.HIToolbox           	0x00007fff31d0f895 RunCurrentEventLoopInMode + 293
30  com.apple.HIToolbox           	0x00007fff31d0f4d4 ReceiveNextEventCommon + 371
31  com.apple.HIToolbox           	0x00007fff31d0f348 _BlockUntilNextEventMatchingListInModeWithFilter + 64
32  com.apple.AppKit              	0x00007fff2ffcc95b _DPSNextEvent + 997
33  com.apple.AppKit              	0x00007fff2ffcb6fa -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1362
34  com.apple.AppKit              	0x00007fff2ffc575d -[NSApplication run] + 699
35  libqcocoa.dylib               	0x000000010582525b 0x1057f8000 + 184923
36  org.qt-project.QtCore         	0x00000001037fe61f QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) + 431
37  org.qt-project.QtCore         	0x00000001038035c2 QCoreApplication::exec() + 130
38  com.vector35.binaryninja      	0x0000000100473637 0x100468000 + 46647
39  libdyld.dylib                 	0x00007fff5fb10085 start + 1

Thread 7 Crashed:
0   libsystem_kernel.dylib        	0x00007fff5fc4eb86 __pthread_kill + 10
1   libsystem_pthread.dylib       	0x00007fff5fd04c50 pthread_kill + 285
2   libsystem_c.dylib             	0x00007fff5fbb81c9 abort + 127
3   libsystem_c.dylib             	0x00007fff5fb80868 __assert_rtn + 320
4   libc++.1.dylib                	0x00007fff5d1e2b74 std::__1::mutex::unlock() + 46
5   libbinaryninjacore.dylib      	0x0000000101753fd6 0x101528000 + 2277334
6   libbinaryninjacore.dylib      	0x00000001017467aa 0x101528000 + 2221994
7   libbinaryninjacore.dylib      	0x00000001016a5910 0x101528000 + 1562896
8   libbinaryninjacore.dylib      	0x00000001016a862d 0x101528000 + 1574445
9   libsystem_pthread.dylib       	0x00007fff5fd0233d _pthread_body + 126
10  libsystem_pthread.dylib       	0x00007fff5fd052a7 _pthread_start + 70
11  libsystem_pthread.dylib       	0x00007fff5fd01425 thread_start + 13
```

**Comment by bpotchik** [comment]
This is a crash in LogWindow. I believe I have a fix for this locally. Will push soon.

**Comment by bpotchik** [comment]
Fixed in build 1.1.1496-dev.
